lib/rules/no-return-await.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * @fileoverview Disallows unnecessary `return await`
+ * @author Jordan Harband
+ */
+"use strict";
+
+const functionTypes = new Set([
+    "FunctionExpression",
+    "FunctionDeclaration",
+    "ArrowFunctionExpression",
+    "MethodDefinition",
+]);
+
+/**
+ * @param {ASTNode} node the node from which to locate a containing function
+ * @returns {ASTNode} the containing function node
+ */
+function getContainingFunction(node) {
+    if (!node.parent || functionTypes.has(node.type)) {
+        return node;
+    }
+    return getContainingFunction(node.parent);
+}
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+const message = "Redundant use of `await` on a return value.";
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "disallow unnecessary `return await`",
+            category: "Best Practices",
+            recommended: false // TODO: set to true
+        },
+        fixable: false,
+        schema: [
+        ]
+    },
+
+    create(context) {
+        return {
+            ArrowFunctionExpression(node) {
+                if (node.async && node.body.type === "AwaitExpression") {
+                    const sourceCode = context.getSourceCode();
+                    const loc = node.body.loc;
+
+                    context.report({
+                        node: sourceCode.getFirstToken(node.body),
+                        loc,
+                        message,
+                    });
+                }
+            },
+            ReturnStatement(node) {
+                const argument = node.argument;
+
+                if (argument && argument.type === "AwaitExpression") {
+                    const sourceCode = context.getSourceCode();
+                    const loc = argument.loc;
+                    const containingFunction = getContainingFunction(node);
+
+                    if (containingFunction.type !== "ArrowFunctionExpression" || containingFunction.body.type === "BlockStatement") {
```

Comment:
I think this check can be removed; it applies to every return statement (since arrow functions with implicit return don't actually contain a `ReturnStatement`).